### DISCUSSION TITLE
Is most frequent language calculation based on the file size?

### DISCUSSION THREAD

**Discussion by iamgiolaga** [anuraghazra/github-readme-stats#3518] [Q&A] [answered]
### Is your feature request related to a problem? Please describe.

Most frequent language in my case is cleary unbalanced when I add Jupyter Notebook to my profile. Being them heavy files, they represent the majority of my usages, even if they represent the minority in terms of file number and lines of code.

### Describe the solution you'd like

I would like a weighted mean calculation that doesn't weight that much the file size, otherwise languages with light files will always tend to represent the minor percentage.

### Additional context

<img width="344" alt="Screenshot 2023-12-01 at 09 47 43" src="https://github.com/anuraghazra/github-readme-stats/assets/67608220/6e464c10-42e6-4ce5-82d3-a4064d57d8ae">

**Comment by qwerty541** [comment, marked as answer]
Hey, Giovanni! You are able to calibrate language calculation algorithm with `&size_weight=0.5` and `&count_weight=0.5` querystring parameters depending on your purposes. More information about language calculation algorithm can be found there: https://github.com/anuraghazra/github-readme-stats#language-stats-algorithm

### Before

```html
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=iamgiolaga&layout=compact&langs_count=8&theme=tokyonight" />
```

<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=iamgiolaga&layout=compact&langs_count=8&theme=tokyonight" />

### After

```html
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=iamgiolaga&layout=compact&langs_count=8&theme=tokyonight&size_weight=0.5&count_weight=0.5" />
```

<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=iamgiolaga&layout=compact&langs_count=8&theme=tokyonight&size_weight=0.5&count_weight=0.5" />

**Reply by rickstaa** [reply]
@iamgiolaga In addition to your @qwerty541's points, the current language card has a few known bugs that might lead to inaccuracies in the language results for users with over 100 repositories (check out the details here: https://github.com/anuraghazra/github-readme-stats/issues/1852). I plan to address this by rolling out a [GitHub action](https://github.com/anuraghazra/github-readme-stats/issues/2179) in the first quarter of the upcoming year. Also, our language calculations are based on repository language results (more details at https://github.com/anuraghazra/github-readme-stats/issues/1801) due to GitHub's GraphQL API limitations. If you're keen on enhancing the language results, your support would be greatly appreciated at https://github.com/orgs/community/discussions/18230 🙏🏻. Thanks!

**Reply by iamgiolaga** [reply]
Thanks for your clear answers guys @qwerty541 @rickstaa 👏🏻
